### PR TITLE
[DSCP-192] Solve NAT-related issue, add host resolving

### PR DESCRIPTION
**Pull request by volhovm**

### CODE REVIEW

File: witness/src/Dscp/Resource/Network.hs [hunk 1]
```diff
@@ -83,8 +84,10 @@ instance WithNetLogging => AllocResource NetCliParams NetCliResources where
         buildComponentR "netcli" allocate release
       where
         allocate = do
+            ncPeers' <- liftIO $ Set.fromList <$> mapM mkZTNodeId (Set.toList ncPeers)
```
Comment (Heimdell) [inline witness/src/Dscp/Resource/Network.hs]: Had to look in to recognise that this is a hand-made `mapM` on `Set`.

I'd make `mapMSet` function somewhere and move it later to some `*.Util`.

File: witness/src/Dscp/Resource/Network.hs [hunk 2]
```diff
@@ -123,9 +126,15 @@ instance WithNetLogging => AllocResource NetServParams NetServResources where
         buildComponentR "netcli" allocate release
       where
         allocate = do
+            nsPeers' <- liftIO $ Set.fromList <$> mapM mkZTNodeId (Set.toList nsPeers)
+            nsOurAddress' <- liftIO $ mkZTNodeId nsOurAddress
+            nsInternalAddress' <-
+                liftIO $ maybe (pure Nothing) (fmap Just . mkZTNodeId) nsInternalAddress
```
Comment (Heimdell) [inline witness/src/Dscp/Resource/Network.hs]: Isn't `maybe (... Nothing) (... Just ...)` a `fmap x`? Or, maybe whole expression can be written as `(fmap.fmap) mkZTNodeId nsInternalAddress`?

File: witness/src/Dscp/Resource/Network.hs [hunk 3]
```diff
@@ -123,9 +126,15 @@ instance WithNetLogging => AllocResource NetServParams NetServResources where
...
+
             global <- ztGlobalEnv (unNetLogging netLogging)
-            cli <- createNetCliEnv global nsPeers
-            serv <- createNetServEnv global nsOurAddress nsInternalAddress
+            cli <- createNetCliEnv global nsPeers'
+            serv <- createNetServEnv global nsOurAddress' nsInternalAddress'
```
Comment (Heimdell) [inline witness/src/Dscp/Resource/Network.hs]: Also, lets not use names with `'`. Its easy to forgot `'` and get yourself incorrect behaviour or bottom and an hour of wtf-debug.

Lets call them `new-` and `old-` instead. Or `src-`/`tmp-`/`dst-`.

Comment (volhovm) [reply witness/src/Dscp/Resource/Network.hs]: ok

**Pull request merged**